File: Algorithms/C/LinearSearch.c
// Program to search an element in an array using Linear Search.

#include <stdio.h>
#include <stdlib.h>

// Function to scan an array as an input.

void scanArray(int array[], int size) {

     for(int i=0; i<size; i++) 
          scanf("%d", &array[i]);
          
}

// Function to implement Linear Search.

int linearSearch(int array[], int size, int key) {

     int res=-1;                   // If element is not in the given array -1 will be returned.
     for(int i=0; i<size; i++) {
          if(array[i]==key)
               res=i;              // Result will be the index where the element if found. 
                       
     }
     return res;
}

// Main Function

int main() {

     int array[5]={0};             // Initialising Array with Zero.
     int key;
     int result;
     printf("Enter the elements of array:\n");
     scanArray(array,5);
     printf("Enter the key element :\n");
     scanf("%d", &key);
     result = linearSearch(array,5,key);
     if (result==-1)
          printf("Key element not found!\n");
     else
          printf("Key element is found at index : %d\n", result);
     return 0;
}     

File: Algorithms/Java/linearsearch.java
/* 
Problem: Given an array arr[] of n elements, write a function to search a given element x in arr[].

Examples :

Input : arr[] = {10, 20, 80, 30, 60, 50, 
                     110, 100, 130, 170}
          x = 110
Output : 6
Element x is present at index 6

*/

import java.util.Scanner;  
   
public class linearsearch   
{  
  public static void main(String args[])  
  {  
    int c, n, search, array[];  
   
    Scanner in = new Scanner(System.in);  
    System.out.println("Enter number of elements");  
    n = in.nextInt();   
    array = new int[n];  
   
    System.out.println("Enter those " + n + " elements");  
   
    for (c = 0; c < n; c++)  
      array[c] = in.nextInt();  
   
    System.out.println("Enter value to find");  
    search = in.nextInt();  
   
    for (c = 0; c < n; c++)  
    {  
      if (array[c] == search)     /* Searching element is present */  
      {  
         System.out.println(search + " is present at location " + (c + 1) + ".");  
          break;  
      }  
   }  
   if (c == n)  /* Element to search isn't present */  
      System.out.println(search + " isn't present in array.");  
  }  
}

